client/scripts/controllers/chain/substrate/account.ts

```diff
@@ -129,26 +152,31 @@ class SubstrateAccounts implements IAccountsModule<SubstrateCoin, SubstrateAccou
           stakersCall.multi(currentSet.map((elt) => stakersCallArgs(elt.toString()))),
           api.query.staking.bonded.multi(toBeElected.map((elt) => elt.toString())),
           stakersCall.multi(toBeElected.map((elt) => stakersCallArgs(elt.toString()))),
+          of(validatorsInfo)
         );
       }),
       auditTime(100),
-      map(([currentSet, toBeElected, controllers, exposures, nextUpControllers, nextUpExposures]:
-        [ AccountId[], AccountId[], Vec<AccountId>, Exposure[], Vec<AccountId>, Exposure[] ]) => {
+      map(([currentSet, toBeElected, controllers, exposures, nextUpControllers, nextUpExposures, validatorsInfo]:
+        [ AccountId[], AccountId[], Vec<AccountId>, Exposure[], Vec<AccountId>, Exposure[], IValidatorInfo ]) => {
         const result: IValidators = {};
         for (let i = 0; i < currentSet.length; ++i) {
-          result[currentSet[i].toString()] = {
+          const key = currentSet[i].toString();
+          result[key] = {
             exposure: exposures[i],
             controller: controllers[i].toString(),
             isElected: true,
+            commissionPer: validatorsInfo[key] ? validatorsInfo[key].commissionPer : 0
```

Comment:
Overall seems ok, just want to understand when commission or key wouldn't exist in the map.

Reply:
I was getting undefined randomly. So I added condition for safe property gets.